### PR TITLE
Metrics: Add gauge for requests currently in flight

### PR DESCRIPTION
**Pull request by bergquist**
I placed the metric in request_metrics.go instead of metrics.go.
having all metrics in one place have become messy I think. WDYT?

### CODE REVIEW

**Review by aknuds1 — APPROVED** [review]
LGTM

**Review by marefr — APPROVED** [review]
Looks good. But wonder how much this would help given that Prometheus scrapes each 10 or 15 seconds. Wouldn't that mean that Prometheus only get the value of the gauge each 10/15 seconds?

File: pkg/middleware/request_metrics.go [hunk 1]
```diff
@@ -7,13 +7,32 @@ import (
...
 )
 
+var (
+	httpRequestsInFlight prometheus.Gauge
+)
+
+func init() {
+	httpRequestsInFlight = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Name: "http_requests_in_flight",
```
Comment (marefr) [inline pkg/middleware/request_metrics.go]: To follow the naming of the other http_request_ metrics I suggest the following change:

```suggestion
			Name: "http_request_in_progress",
```

Comment (aknuds1) [reply pkg/middleware/request_metrics.go]: @marefr Why rename in_flight to in_progress? I think they are synonymous? Renaming from http_requests_ to http_request is to be consistent, AFAICT, even though I'm not familiar with the history.

Comment (marefr) [reply pkg/middleware/request_metrics.go]: Personal opinion. "in flight" makes me think about flights/flying, but maybe that's only me?

Comment (aknuds1) [reply pkg/middleware/request_metrics.go]: I think "in flight" is a normal term to refer to for example requests being handled. You can see [here](https://english.stackexchange.com/a/189987) for a discussion.

Comment (marefr) [reply pkg/middleware/request_metrics.go]: Okay thanks :+1:

Comment (marefr) [reply pkg/middleware/request_metrics.go]: Renamed to `http_request_in_flight`.

**Comment by SuperQ** [comment]
@marefr These kinds of gauges can be useful to find stuck-open requests and can be interesting for services under high load.

**Comment by marefr** [comment]
@SuperQ yeah after some discussions I came to that same understanding. Thanks

**Comment by SuperQ** [comment]
My only nit is I would usually name this metric `http_active_requests`. :grinning:

**Comment by bergquist** [comment]
I got "in_flight" from https://github.com/prometheus/client_golang/blob/master/prometheus/promhttp/http.go#L257 :)

**Pull request merged**